server/src/main/java/org/elasticsearch/index/mapper/PassThroughObjectMapper.java

```diff
@@ -61,42 +81,66 @@ public PassThroughObjectMapper build(MapperBuilderContext context) {
                 enabled,
                 dynamic,
                 buildMappers(context.createChildContext(name(), timeSeriesDimensionSubFields.value(), dynamic)),
-                timeSeriesDimensionSubFields
+                timeSeriesDimensionSubFields,
+                supersededBy
             );
         }
     }
 
     // If set, its subfields are marked as time-series dimensions (for the types supporting this).
     private final Explicit<Boolean> timeSeriesDimensionSubFields;
 
+    private final Set<String> supersededBy;
```

Comment:
Shouldn't this just be `private final String supersededBy;` ?

Reply:
Thanks, it would be good to check whether this is really is required. This does increase complexity and we shouldn't do this without a concrete need for this.